add recipe for responsive sidebar (#2002)
slint-ui/slint: docs/recipes/recipes.md

}
```

### Breakpoints for responsive ui

> Uses this recipe to implement a responsive SideBar that collapses when the parent width is smaller then the given break-point. On Button
> click the SideBar can be expanded if collapsed on break-point.

Reviewer: ```suggestion
Use this recipe to implement a responsive SideBar that collapses when the parent width is smaller than the given break-point. When the button is clicked, the SideBar expands if collapsed on break-point.
```

```slint,no-auto-preview
import { Button } from "std-widgets.slint";

export SideBar := Rectangle {
    private property <bool> collapsed: reference-width < break-point;
    property <length> reference-width;

>     property <length> break-point: 600px;

Reviewer: Since this is copy & paste ready, I wonder if it should have basically "API" documentation via comments here, and comments down at the use-site how it is used.

Perhaps the text of the button should be a configurable property?

That said, I guess this needs editing anyway as users may want to add more than just one button (like you did in your clip).

Author: It's a good suggestion. I will improve it. Thanks.

    width: 160px;
   
    container := Rectangle {   
        private property <bool> expaned;
        width: parent.width;
        background: black;

        VerticalLayout { 
            padding: 4px;
            alignment: start;
            if (collapsed) : Button {
                checked: container.expaned;
                text: "C";

                clicked => {  
                    container.expaned = !container.expaned;

>                 }

Reviewer: I tried in the online editor, and the "C" is padded a bit too far, I think:

<img width="156" alt="Screenshot 2022-12-16 at 12 14 25" src="https://user-images.githubusercontent.com/1486/208086566-13c15def-7ef7-47ff-aed6-30709c9ee34e.png">
